src/main/java/fitnesse/wikitext/widgets/MavenArtifactClasspathExtractor.java

```diff
@@ -0,0 +1,134 @@
+        }
+    }
+
+    private List<String> parseClasspath(File dependencyList) throws MavenArtifactClasspathExtractionException {
+        Pattern pattern = Pattern.compile("(.*):([^:]+)");
+        BufferedReader reader = null;
+        try {
+            reader = new BufferedReader(new FileReader(dependencyList));
+            while (!"The following files have been resolved:".equals(reader.readLine())) ;
+            List<String> classpath = new ArrayList<String>();
+            String line;
+            while ((line = reader.readLine()) != null && !line.isEmpty()) {
+                Matcher matcher = pattern.matcher(line);
+                if (matcher.find()) {
+                    // group 1 contains the path to the file
+                    classpath.add(matcher.group(2));
+                }
+            }
+            return classpath;
+        } catch (IOException e) {
+            throw new MavenArtifactClasspathExtractionException("Could not extract classpath", e);
+        } finally {
+            if(reader!=null) {
+                try {
+                    reader.close();
+                } catch (IOException e) {
+                    throw new MavenArtifactClasspathExtractionException("Could not extract classpath", e);
+                }
+            }
+        }
+    }
+
+    private File invokeDependencyList(File pomFile) throws IOException, MavenArtifactClasspathExtractionException {
+        DefaultInvocationRequest request = new DefaultInvocationRequest();
+        request.setMavenOpts(System.getProperty("MAVEN_OPTS"));
+        request.setUserSettingsFile(new File(System.getProperty("MAVEN_SETTINGS_PATH")));
+        request.setBatchMode(true);
+        request.setUpdateSnapshots(true);
+        request.setGoals(Arrays.asList("-f " + pomFile.getPath(), "dependency:list"));
+
+        File dependencyList = File.createTempFile("fitnesse-maven-classpath-", ".deps");
+        Properties properties = new Properties();
+        properties.setProperty("outputFile", dependencyList.getPath()); // redirect output to a file
+        properties.setProperty("outputAbsoluteArtifactFilename", "true"); // with paths
+        request.setProperties(properties);
+
+        DefaultInvoker defaultInvoker = new DefaultInvoker();
+//        defaultInvoker.setOutputHandler(null); // not interested in Maven output itself
+        try {
+            InvocationResult result = defaultInvoker.execute(request);
+            if (result.getExitCode() != 0) {
+                throw new IllegalStateException("Build failed.");
+            }
+        } catch (MavenInvocationException e) {
+            throw new MavenArtifactClasspathExtractionException("Could not extract classpath", e);
+        }
+        return dependencyList;
+    }
+
+    private File createPomFile(MavenArtifactClasspathSymbolType.ParsedSymbol parsedSymbol) throws IOException {
+        File pomFile = File.createTempFile("fitnesse-maven-classpath-", ".pom");
```

Comment:
Isn't there a way to obtain a resolver and get the dependency directly? Creating a temp file does not look like the right solution for this.